web/src/app/pages/create-survey/data-sharing-terms/data-sharing-terms.component.spec.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2024 The Ground Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CommonModule} from '@angular/common';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+
+import {DataSharingType} from 'app/models/survey.model';
+import {DataSharingTermsComponent} from 'app/pages/create-survey/data-sharing-terms/data-sharing-terms.component';
+
+describe('DataSharingTermsComponent', () => {
+  let component: DataSharingTermsComponent;
+  let fixture: ComponentFixture<DataSharingTermsComponent>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [DataSharingTermsComponent],
+      imports: [CommonModule],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(DataSharingTermsComponent);
+    fixture.componentInstance.type = DataSharingType.PUBLIC;
+    fixture.componentInstance.customText = 'Hey there here is an agreement';
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('displays mat-cards for each of the three data sharing types', () => {
+    const matCards: HTMLElement[] =
```

Comment:
For the material components, it's highly recommended to use the component harnesses. Here is the documentation for card: https://material.angular.io/components/card/api#MatCardHarness as well as an example: https://material.angular.io/components/card/examples#card-harness.

I think it's reasonable to make this a separate issue to do in a follow up PR after the code freeze since this is a test.

Reply:
Yeah, good call, will do this in a follow-up to get this merged before the freeze